server/region_handler.go

```diff
@@ -0,0 +1,512 @@
+		return nil, errors.Trace(err)
+	}
+	tableID := table.Meta().ID
+
+	tableRegions = &TableRegions{
+		TableName: tableName,
+		TableID:   tableID,
+		Indices:   make([]IndexRegions, len(table.Indices())),
+	}
+
+	// for primary
+	startKey, endKey := getTableHandleKeyRange(tableID)
+	tableRegions.RowRegions, err = req.regionCache.ListRegionIDsInKeyRange(req.bo, startKey, endKey)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// for indexes
+	for i, index := range table.Indices() {
+		indexID := index.Meta().ID
+		tableRegions.Indices[i].Name = index.Meta().Name.String()
+		tableRegions.Indices[i].ID = indexID
+		startKey, endKey := getTableIndexKeyRange(tableID, indexID)
+		tableRegions.Indices[i].Regions, err = req.regionCache.ListRegionIDsInKeyRange(req.bo, startKey, endKey)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	return tableRegions, nil
+}
+
+func (req *RegionsHTTPRequest) getRegionByID(regionID uint64) (*RegionItem, error) {
+	region, err := req.regionCache.LocateRegionByID(req.bo, regionID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	indexRange := NewRegionIndexRange(region)
+
+	regionItem := &RegionItem{
+		RegionID: regionID,
+		StartKey: region.StartKey,
+		EndKey:   region.EndKey,
+		Indices:  []IndexItem{},
+	}
+
+	// table's number is smaller than 1000, iterate table ID in index range.
+	if indexRange.lastTableID()-indexRange.firstTableID() <= 1000 {
+		for tableID := indexRange.firstTableID(); tableID <= indexRange.lastTableID(); tableID++ {
+			curTable, exist := req.domain.InfoSchema().TableByID(tableID)
+			if exist {
+				start, end := indexRange.getLastInxIDRange()
+				regionItem.addTableIndicesInRange(curTable, start, end)
+			}
+		}
+		return regionItem, nil
+	}
+
+	for _, db := range req.domain.InfoSchema().AllSchemaNames() {
+		for _, table := range req.domain.InfoSchema().SchemaTables(model.NewCIStr(db)) {
+			curTable, exist := req.domain.InfoSchema().TableByID(table.Meta().ID)
+			if exist {
+				start, end := indexRange.getLastInxIDRange()
+				regionItem.addTableIndicesInRange(curTable, start, end)
+			}
+		}
+	}
+	return regionItem, nil
+}
+
+// RegionsResponse is the response struct for regions.
+type RegionsResponse struct {
+	Success   bool        `json:"status"`
+	Msg       string      `json:"message,omitempty"`
+	RequestID string      `json:"request_id"`
+	Data      interface{} `json:"data,omitempty"`
+}
+
+// NewRegionsResponseWithError create a response item with error
+func NewRegionsResponseWithError(reqID string, err error) *RegionsResponse {
+	return &RegionsResponse{
+		Success:   false,
+		Msg:       err.Error(),
+		RequestID: reqID,
+	}
+}
+
+// NewRegionsResponseWithData create a response item with data
+func NewRegionsResponseWithData(reqID string, data interface{}) *RegionsResponse {
+	return &RegionsResponse{
+		Success:   true,
+		RequestID: reqID,
+		Data:      data,
+	}
+}
+
+func getTableHandleKeyRange(tableID int64) (startKey, endKey []byte) {
```

Comment:
can we move getTableHandleKeyRange and getTableIndexKeyRange to the tablecodec package?